samples/template/src/main.c

```diff
@@ -0,0 +1,517 @@
+static void state_connecting_entry(void *o);
+static void state_connecting_run(void *o);
+
+static const struct smf_state app_states[] = {
+	[STATE_RUNNING] = SMF_CREATE_STATE(NULL, state_running_run, NULL, NULL),
+	[STATE_INIT] = SMF_CREATE_STATE(state_init_entry, state_init_run, NULL,
+					&app_states[STATE_RUNNING]),
+	[STATE_SIDEWALK_READY] =
+		SMF_CREATE_STATE(NULL, state_ready_run, NULL, &app_states[STATE_RUNNING]),
+	[STATE_SIDEWALK_NOT_READY] =
+		SMF_CREATE_STATE(NULL, state_not_ready_run, NULL, &app_states[STATE_RUNNING]),
+	[STATE_SIDEWALK_CONNECTING] = SMF_CREATE_STATE(state_connecting_entry, state_connecting_run,
+						       NULL, &app_states[STATE_RUNNING]),
+};
+
+static uint8_t echo_payload[255];
+struct app_msg app_send_buffer;
+
+uint8_t __aligned(4) msgq_buffer[CONFIG_SIDEWALK_THREAD_QUEUE_SIZE * sizeof(app_event_t)];
+
+static struct sid_ctx_s sid_ctx;
+
+static void on_sidewalk_event(bool in_isr, void *context)
+{
+	LOG_INF("%s", __func__);
+
+	int err = app_event_send(APP_EVENT_SIDEWALK);
+	if (err) {
+		LOG_ERR("send event err %d", err);
+	};
+}
+
+static void on_sidewalk_msg_received(const struct sid_msg_desc *msg_desc, const struct sid_msg *msg,
+				     void *context)
+{
+	LOG_INF("%s", __func__);
+
+	LOG_DBG("received message(type: %d, link_mode: %d, id: %u size %u)", (int)msg_desc->type,
+		(int)msg_desc->link_mode, msg_desc->id, msg->size);
+	LOG_HEXDUMP_INF((uint8_t *)msg->data, msg->size, "Message data: ");
+
+	memcpy(echo_payload, msg->data, msg->size);
+	app_send_buffer = (struct app_msg) {
+			.msg = {
+				.data = echo_payload,
+				.size = msg->size,
+			},
+			.desc = {
+				.type = SID_MSG_TYPE_NOTIFY,
+				.link_type = SID_LINK_TYPE_ANY,
+				.link_mode = SID_LINK_MODE_CLOUD,
+			},
+		};
+
+	int err = app_event_send(APP_EVENT_SEND);
+	if (err) {
+		LOG_ERR("send event err %d", err);
+	};
+}
+
+static void on_sidewalk_msg_sent(const struct sid_msg_desc *msg_desc, void *context)
+{
+	LOG_INF("%s", __func__);
+	LOG_INF("sent message(type: %d, id: %u)", (int)msg_desc->type, msg_desc->id);
+}
+
+static void on_sidewalk_send_error(sid_error_t error, const struct sid_msg_desc *msg_desc,
+				   void *context)
+{
+	LOG_INF("%s", __func__);
+	LOG_ERR("failed to send message(type: %d, id: %u), err:%d", (int)msg_desc->type,
+		msg_desc->id, (int)error);
+}
+
+static void on_sidewalk_factory_reset(void *context)
+{
+	LOG_INF("%s", __func__);
+
+	ARG_UNUSED(context);
+
+	LOG_INF("factory reset notification received from sid api");
+	if (sid_hal_reset(SID_HAL_RESET_NORMAL)) {
+		LOG_WRN("Reboot type not supported");
+	}
+}
+
+static void on_sidewalk_status_changed(const struct sid_status *status, void *context)
+{
+	LOG_INF("%s", __func__);
+	int err_event = 0;
+	int err_led = 0;
+
+	switch (status->state) {
+	case SID_STATE_READY:
+	case SID_STATE_SECURE_CHANNEL_READY:
+		err_event = app_event_send(APP_EVENT_READY);
+		err_led = dk_set_leds_state(DK_LED1_MSK | DK_LED2_MSK, DK_ALL_LEDS_MSK);
```

Comment:
We shouldn't manipulate the LEDs inside the sidewalk callback, it would be better to do inside the applicaiton SMF

Reply:
outdated - moved to state notifier